src/shogun/mathematics/graph/Tensor.h

```diff
@@ -24,96 +25,104 @@ namespace shogun
 {
 	namespace graph
 	{
+		/* Creates a copy of the data or a owning data manager on
+		 * the device
+		 */
+		inline std::shared_ptr<ShogunStorage> device_put(
+		    void* ptr, const Shape& shape,
+		    const std::shared_ptr<NumberType>& type, const bool copy)
+		{
+			return std::make_shared<ShogunStorage>(
+			    ptr, shape, type, copy, ShogunStorage::Copy{});
+		}
+
+		/* Creates a view to a pointer. This that the devide
+		 * memory manager will not call the destructor.
+		 */
+		inline std::shared_ptr<ShogunStorage> device_view(
+		    void* ptr, const Shape& shape,
+		    const std::shared_ptr<NumberType>& type)
+		{
+			return ShogunStorage::create_view(ptr, shape, type);
+		}
```

Comment:
These 2 functions create the appropriate data manager, i.e. whether the manager deallocates on destruction, new memory is allocated or data is moved :)